### PR TITLE
codeintel: prioritize manual autoindexing jobs

### PR DESCRIPTION
**Pull request by Strum355**
Prioritizes manually enqueued autoindexing jobs over background-scheduled jobs based on the enqueuer user ID (derived from the context's actor). This way, the users jobs will get indexed first, instead of being thrown to the back of the queue

## Test plan

Added unit tests to ensure dequeue order and that user ID is properly set from actor

### CODE REVIEW

File: internal/codeintel/autoindexing/internal/background/dependencies/utils.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ var IndexWorkerStoreOptions = dbworkerstore.Options[uploadsshared.Index]{
 	ViewName:          "lsif_indexes_with_repository_name u",
 	ColumnExpressions: indexColumnsWithNullRank,
 	Scan:              dbworkerstore.BuildWorkerScan(scanIndex),
-	OrderByExpression: sqlf.Sprintf("u.queued_at, u.id"),
```
[the remark this answers is not in the document]
Comment (efritz) [reply internal/codeintel/autoindexing/internal/background/dependencies/utils.go:32]: damn you made one nvm

File: migrations/frontend/1691759644_autoindexing_manual_priority_index/up.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+CREATE INDEX CONCURRENTLY IF NOT EXISTS
+lsif_indexes_dequeue_order_idx ON lsif_indexes
+USING btree ((enqueuer_user_id = 0), queued_at DESC, id);
```
Comment (efritz) [inline migrations/frontend/1691759644_autoindexing_manual_priority_index/up.sql:3]: This could be partial on `state = queued` I think.

**Review by efritz — APPROVED** [review]

File: internal/codeintel/autoindexing/internal/background/dependencies/utils.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Strum355) [reply internal/codeintel/autoindexing/internal/background/dependencies/utils.go:32]: :firesnail:

File: migrations/frontend/1691759644_autoindexing_manual_priority_index/up.sql (hunk 2 above)
Comment (Strum355) [reply migrations/frontend/1691759644_autoindexing_manual_priority_index/up.sql:3]: great idea compadre

Comment (Strum355) [reply migrations/frontend/1691759644_autoindexing_manual_priority_index/up.sql]: Updated https://github.com/sourcegraph/sourcegraph/pull/55745/commits/727d4115e2e3598eb8d3a8db17fc876e38084206

**Pull request merged**